### PR TITLE
User friendly UI for defining offer parameters

### LINKED ISSUE #1388: Make easier the way the Service provider add the offer parameters in Backoffice
**Is your feature request related to a problem? Please describe.**
Now, when the service provider wants to add the service offer parameters from backoffice he needs to add them in JSON format:  
We have the feedback that it is hard for standard people to follow this procedure. 

**Describe the solution you'd like**
We would like to add the possibility to type the service parameters for the user in an easier way. 
We have to think about how to deal with it in a clever way using the current implementation of the attributes.
All possible types and options of attributes are described here:
https://confluence.egi.eu/pages/viewpage.action?pageId=37454529

**Additional context**
@abacz will deliver prev

### LINKED ISSUE #1430: Graphic preview for defining offer parameters
My proposition to discuss - scenario:
First 3 steps are clickable invision prototype (try it!) ;)
1. User fills the form 
https://invis.io/3BWDZ1JTH5R#/409065926_Backoffice_Service_Item_-_Add_New_Offer_-_Empty
2. User chooses parameter type (here select) and "add arrow" appears
https://invis.io/72WDZ2FBDHS#/409065925_Backoffice_Service_Item_-_Add_New_Offer_-_Select
3. User clicks arrow button and select box is ready to fill
https://invis.io/K9WDZ4UG5R8#/409065924_Backoffice_Service_Item_-_Show_Select_Parameter_Box
4. User can add few more parameter boxes. Left section "parameter type to add" is always visible (fixed) in offer parameters area.
https://invis.io/BGWDZ7KK76A#/409065923_Backoffice_Service_Item_-_Fwa_Parameters_Boxes

I'am not sure what kind of inputs we should have in every parameter type. 
@mkasztelnik, @agpul, @martaswiatkowska, @bwilk

### PR DESCRIPTION
**Pull request by mkasztelnik**
This PR will be a result of a collaboration effort of @mkasztelnik, @martaswiatkowska, @goreck888, and @abacz to deliver user-friendly UI for managing  offer attributes. The scope includes:
  * [x] core mechanism for storing parameter templates and the possibility to add/remove it in an automatic way (@mkasztelnik)
  * [x] drawings showing how parameter forms should looks like (@abacz) #1430 
  * [x] Improve view like on graphic previews (@jarekzet) #1444
  * [x] possibility to move parameter up/down (@mkasztelnik)
  * [x] parameter validation mechanism (@mkasztelnik)
  * [x] parameter create/update mechiamism (@mkasztelnik)
  * [x] parameter templates for:
    * [x] date (@goreck888)
    * [x] input (@mkasztelnik)
    * [x] select (@martaswiatkowska)
    * [x] multiselect (have difference with minItems and maxItems property comparing to select) (@martaswiatkowska)
    * [x] range (@mkasztelnik)
    * [x] quantity price (@goreck888)
  * [x] update specs (@goreck888, @mkasztelnik)
  * [x] ~~#1447 create migration which updates quantity price attribute type from `quantity-price` to `quantity_price` (@goreck888)~~
  
Fixes #1388 
Fixes #1430

### CODE REVIEW

**Comment by agpul** [comment]
Can you fix this PR? I click "Add new Offer" and I've got: 
image
I would like to have the possibility to test this PR to create the ticket for the next sprint.

**Comment by agpul** [comment]
1. The first comment is related to UI:
image
For me, it looks like this arrow is unavailable in this view.

**Comment by MinLidia** [comment]
I suggest to move button to the right. It took mi time to fond where I need to click to save changes 
Screenshot 2020-03-27 at 15 30 34

**Comment by jarekzet** [comment]
added last CSS styling fixes, collapse cards

**Comment by jarekzet** [comment]
added css mods:
1. Sticky left column
2. Card background color
3. validation icons positioning

**Comment by agpul** [comment]
I suggest change from +Add to Type somethig ? @MinLidia @abacz 
image

**Comment by abacz** [comment]
> I suggest change from +Add to Type somethig ? @MinLidia @abacz

start typing to add?

**Comment by agpul** [comment]
> > I suggest change from +Add to Type somethig ? @MinLidia @abacz
> 
> start typing to add?

agree

**Comment by MinLidia** [comment]
One more request for change : 

To reproduce 1. Add one parametr 2. Collapse it 3. Add new Parametr 


Screenshot 2020-03-30 at 14 00 19

File: app/javascript/stylesheets/_bootstrap-customizations.scss [hunk 1]
```diff
@@ -1007,7 +1007,7 @@ Card style
 
       .card-text {
         padding: 0.8rem;
-        background-color: $white;
+        background-color: $home-card-bg;
```
Comment (abacz) [inline app/javascript/stylesheets/_bootstrap-customizations.scss]: we need to think about variables name

**Pull request merged**